src/main/java/com/merge/api/core/ClientOptions.java

```diff
@@ -95,11 +98,20 @@ public Builder addHeader(String key, Supplier<String> value) {
             return this;
         }
 
+        public Builder addOkHttpConfigurer(Consumer<OkHttpClient.Builder> okHttpConfigurer) {
+            this.okHttpConfigurer = okHttpConfigurer;
+            return this;
+        }
+
         public ClientOptions build() {
-            OkHttpClient okhttpClient = new OkHttpClient.Builder()
-                    .addInterceptor(new RetryInterceptor(3))
-                    .build();
-            return new ClientOptions(environment, headers, headerSuppliers, okhttpClient);
+            OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder()
+                    .addInterceptor(new RetryInterceptor(3));
```

Comment:
Shouldn't it be configurable as well in case I want to implement my own retry logic? And this one as a default retry mechanism that can be replaced if needed

Reply:
In theory you could add a new interceptor that handles retry in your way, but I'm not sure if you could clear any existing interceptors already, but you might be able to if this got merged.

Don't hold your breath though @bitsal, there's been no movement on this PR or acknowledgement of this from the merge team since I've raised the idea.